File: src/main/java/eu/karelhovorka/zpevnik/formatter/SongFormatter.java
package eu.karelhovorka.zpevnik.formatter;


import eu.karelhovorka.zpevnik.text.SectionTokenizer;
import eu.karelhovorka.zpevnik.text.SongText;

public abstract class SongFormatter {
    public static final String CHORD_URL_PROTOCOL = "chord://";

    public abstract String formatHtml(SongText context);

    protected abstract String formatHtmlChords(String content);

    public static boolean isValidSectionFormat(String content) {
        return SectionTokenizer.isValid(content);
    }
}

File: src/commonMain/kotlin/eu/karelhovorka/zpevnik/formatter/SongFormatter.kt
package eu.karelhovorka.zpevnik.formatter


import eu.karelhovorka.zpevnik.text.SectionTokenizer
import eu.karelhovorka.zpevnik.text.SongText
import kotlin.jvm.JvmField

abstract class SongFormatter {

    abstract fun formatHtml(songText: SongText): String

    protected abstract fun formatHtmlChords(content: String): String

    companion object {

        @JvmField
        val CHORD_URL_PROTOCOL = "chord://"

        fun isValidSectionFormat(content: String): Boolean {
            return SectionTokenizer.isValid(content)
        }
    }
}
